### PR TITLE
Update placeholder list source to support first item in list

### PR DESCRIPTION
**Pull request by MebinAbraham**
### What is the context of this PR?
This PR adds support to placeholder list sources to take into account the first item in the list.

### How to review 
Needed for https://github.com/ONSdigital/eq-questionnaire-schemas/pull/186
Ensure the unit test is as expected.

### Checklist

* [ ] New static content marked up for translation
* [ ] Newly defined schema content included in eq-translations repo

### CODE REVIEW

File: app/questionnaire/placeholder_parser.py [hunk 1]
```diff
@@ -46,9 +47,16 @@ def _resolve_value_source(self, value_source):
         if value_source["source"] == "metadata":
             return self._resolve_metadata_value(value_source["identifier"])
         if value_source["source"] == "list":
-            if value_source.get("id_selector") == "same_name_items":
-                return self._list_store[value_source["identifier"]].same_name_items
-            return len(self._list_store[value_source["identifier"]])
+            id_selector = value_source.get("id_selector")
```
Comment (LJBabbage) [inline app/questionnaire/placeholder_parser.py:50]: I didn't think we were updating runner for this?

Comment (MebinAbraham) [reply app/questionnaire/placeholder_parser.py:50]: Yup, it requires a minor change it runner.

File: app/questionnaire/placeholder_parser.py [hunk 2]
```diff
@@ -46,9 +47,16 @@ def _resolve_value_source(self, value_source):
...
+            list_store: ListModel = self._list_store[value_source["identifier"]]
+
+            if id_selector == "same_name_items":
+                return list_store.same_name_items
+
+            if id_selector == "first":
+                return list_store.first
```
Comment (JamesGardiner) [inline app/questionnaire/placeholder_parser.py:57]:
```suggestion
            if id_selector:
                return getattr(list_store, id_selector)
```

Would ☝️ be more elegant?

**Review by LJBabbage — APPROVED** [review]

**Review by JamesGardiner — APPROVED** [review]

File: app/questionnaire/placeholder_parser.py [hunk 3]
```diff
@@ -46,9 +47,13 @@ def _resolve_value_source(self, value_source):
...
-            if value_source.get("id_selector") == "same_name_items":
-                return self._list_store[value_source["identifier"]].same_name_items
-            return len(self._list_store[value_source["identifier"]])
+            id_selector = value_source.get("id_selector")
+            list_store: ListModel = self._list_store[value_source["identifier"]]
+
+            if id_selector:
+                return getattr(list_store, id_selector)
+
+            return len(list_store)
```
Comment (ajmaddaford) [inline app/questionnaire/placeholder_parser.py:56]:
```suggestion
            list_model: ListModel = self._list_store[value_source["identifier"]]

            if id_selector:
                return getattr(list_model, id_selector)

            return len(list_model)
```

**Pull request merged**